src/test/java/com/training/assignment/tests/AssignmentTest.java

```diff
@@ -165,4 +167,123 @@ public void testAbstractInterfaceConcept() {
+
+            softAssert.assertThat(empDetails.getName())
+                    .as("Name's inside created list and clone list differs")
+                    .isEqualTo(cloneEmpDetails.getName());
+
+            softAssert.assertThat(empDetails.getAddress())
+                    .as("Address inside created list and clone list differs")
+                    .isEqualTo(cloneEmpDetails.getAddress());
+
+            softAssert.assertThat(empDetails.getId())
+                    .as("Id's inside created list and clone list differs")
+                    .isEqualTo(cloneEmpDetails.getId());
+
+            softAssert.assertThat(empDetails.getSalary())
+                    .as("Salary inside created list and clone list differs")
+                    .isEqualTo(cloneEmpDetails.getSalary());
+
+        }
+
+        List<com.training.assignments.miniproject.Employee> deletedEmployee = originalEmployee.getDeletedEmployee(1);
+        List<com.training.assignments.miniproject.Employee> cloneDeletedEmp = cloneEmployee.getCloneDeletedEmployee(1);
+
+        Iterator<com.training.assignments.miniproject.Employee> deletedEmployeeIterator = deletedEmployee.iterator();
+        Iterator<com.training.assignments.miniproject.Employee> cloneDeletedEmployeeIterator = cloneDeletedEmp.iterator();
+
+        while (deletedEmployeeIterator.hasNext() && cloneDeletedEmployeeIterator.hasNext()) {
+            com.training.assignments.miniproject.Employee employee = deletedEmployeeIterator.next();
+            com.training.assignments.miniproject.Employee clonedEmpoyee = cloneDeletedEmployeeIterator.next();
+
+            softAssert.assertThat(employee.getAddress())
+                    .as("Address inside updated list and clone list differs")
+                    .isEqualTo(clonedEmpoyee.getAddress());
+
+            softAssert.assertThat(employee.getName())
+                    .as("Name's inside updated list and clone list differs")
+                    .isEqualTo(clonedEmpoyee.getName());
+
+            softAssert.assertThat(employee.getId())
+                    .as("Id's inside updated list and clone list differs")
+                    .isEqualTo(clonedEmpoyee.getId());
+
+            softAssert.assertThat(employee.getSalary())
+                    .as("Salary inside updated list and clone list differs")
+                    .isEqualTo(clonedEmpoyee.getSalary());
+        }
+
+        List<com.training.assignments.miniproject.Employee> updatedEmployee = originalEmployee.getUpdatedEmployee(0);
+        List<com.training.assignments.miniproject.Employee> cloneUpdatedEmployeeEmp = originalEmployee.getCloneUpdatedEmployee(0);
+
+        Iterator<com.training.assignments.miniproject.Employee> updatedEmployeeIterator = updatedEmployee.iterator();
+        Iterator<com.training.assignments.miniproject.Employee> updatedCloneEmployeeIterator = cloneUpdatedEmployeeEmp.iterator();
```

Comment:
Dont create list updatedEmployee, cloneUpdatedEmployeeEmp, pass function call direclty in .iterator()
  Iterator<com.training.assignments.miniproject.Employee> updatedEmployeeIterator = originalEmployee.getUpdatedEmployee(0).iterator();